homeassistant/components/nws/weather.py

```diff
@@ -0,0 +1,378 @@
+                "Thunderstorm (medium cloud cover)",
+                "Thunderstorm (low cloud cover)",
+            ],
+        ),
+        ("lightning", []),
+        ("pouring", []),
+        (
+            "rainy",
+            [
+                "Rain",
+                "Rain showers (high cloud cover)",
+                "Rain showers (low cloud cover)",
+            ],
+        ),
+        ("windy-variant", ["Mostly cloudy and windy", "Overcast and windy"]),
+        (
+            "windy",
+            [
+                "Fair/clear and windy",
+                "A few clouds and windy",
+                "Partly cloudy and windy",
+            ],
+        ),
+        ("fog", ["Fog/mist"]),
+        ("clear", ["Fair/clear"]),  # sunny and clear-night
+        ("cloudy", ["Mostly cloudy", "Overcast"]),
+        ("partlycloudy", ["A few clouds", "Partly cloudy"]),
+    ]
+)
+
+ERRORS = (aiohttp.ClientError, JSONDecodeError)
+
+FORECAST_MODE = ["daynight", "hourly"]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_LATITUDE): cv.latitude,
+        vol.Optional(CONF_LONGITUDE): cv.longitude,
+        vol.Optional(CONF_MODE, default="daynight"): vol.In(FORECAST_MODE),
+        vol.Optional(CONF_STATION): cv.string,
+        vol.Required(CONF_API_KEY): cv.string,
+    }
+)
+
+
+def convert_condition(time, weather):
+    """
+    Convert NWS codes to HA condition.
+
+    Choose first condition in CONDITION_CLASSES that exists in weather code.
+    If no match is found, return first condition from NWS
+    """
+    conditions = [w[0] for w in weather]
+    prec_probs = [w[1] or 0 for w in weather]
+
+    # Choose condition with highest priority.
+    cond = next(
+        (
+            key
+            for key, value in CONDITION_CLASSES.items()
+            if any(condition in value for condition in conditions)
+        ),
+        conditions[0],
+    )
+
+    if cond == "clear":
+        if time == "day":
+            return "sunny", max(prec_probs)
+        if time == "night":
+            return "clear-night", max(prec_probs)
+    return cond, max(prec_probs)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the NWS weather platform."""
+    from pynws import SimpleNWS
+
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    station = config.get(CONF_STATION)
+    api_key = config[CONF_API_KEY]
+    mode = config[CONF_MODE]
+
+    if None in (latitude, longitude):
+        _LOGGER.error("Latitude/longitude not set in Home Assistant config")
+        return ConfigEntryNotReady
+
+    websession = async_get_clientsession(hass)
+    # ID request as being from HA, pynws prepends the api_key in addition
+    api_key_ha = f"{api_key} homeassistant"
+    nws = SimpleNWS(latitude, longitude, api_key_ha, mode, websession)
```

Comment:
We should check that the api key is valid, and if it's not, log an error and `return`.

Reply:
The API can be any string, although an email address is requested for usage monitoring purposes, I presume.  This is in the docs.  I additionally used the request from HA, as this seems to meet the spirit of the request.  I'm okay taking the whole thing out too since it would still follow their guideline in the strict sense.

Reply:
Ok.